internal/clientconn/conn.go

```diff
@@ -331,7 +332,12 @@ func (c *conn) run(ctx context.Context) (err error) {
 			}
 
 			if c.l.Enabled(ctx, diffLogLevel) {
-				c.l.Log(ctx, diffLogLevel, "Header diff:\n"+diffHeader+"\nBody diff:\n"+diffBody)
+				if len(diffBody) > 0 {
+					diffBody = strings.TrimSpace(diffBody)
+					diffBody = "\n" + diffBody
+				}
+
+				c.l.Log(ctx, diffLogLevel, "Header diff:\n"+diffHeader+"\nBody diff:"+diffBody)
```

Comment:
This is just a thought, the complexity of this code has been bothering me. Can this be any simpler? :thinking: I didn't find a way myself.

Reply:
```go
			if c.l.Enabled(ctx, diffLogLevel) {
				msg := []string{"Header diff:", strings.TrimSpace(diffHeader), "Body diff:"}
				if len(diffBody) > 0 {
					msg = append(msg, strings.TrimSpace(diffBody))
				}

				c.l.Log(ctx, diffLogLevel, strings.Join(msg, "\n"))
			}
```

something like that?

Reply:
That still uses `strings.TrimSpace` excessively.. :sweat_smile: Let's keep what we have.